CRL support fixes from Rustls integration progress

cpu opened the pull request:
This branch is a WIP collection of small things I've noticed to fix as I've started to integrate the webpki CRL support into Rustls.

**Note: left as a draft since I expect there might be other changes to roll in.**

### crl: remove unnecessary unused clippy allows.
These were meant to be removed in earlier work and were overlooked.

### verify_cert: remove CrlProvider, pass CRLs directly.
The existing approach to providing CRLs for use during client certificate validation was over-complicated.

1. The `RevocationCheckOptions` struct wasn't useful, it just wrapped a `CrlProvider` that could have been provided directly.
2. The `CrlProvider` approach creates complicated lifetime considerations in downstream code. What we want is for the CRLs provided to have a lifetime equal to the lifetime of the client certificate validation. The best way to express this is to have the CRLs be provided as arguments to the validation process. This   guarantees the lifetimes we expect.

With these changes in place we can also simplify the generated unit tests. Notably the "Chain depth" notion fell away in previous refactoring but we implemented it in the `CrlProvider` as an example. This isn't necessary now because the semantics are much clearer: if you don't want to check revocation status for intermediates, don't provide a CRL for the intermediates. Ultimately test coverage remains the same but we can drop some tests that were specific to the notion of chain depth. 

### der/crl: fix support for large CRL DER.
Previously we updated the der module to have methods that allow specifying an explicit size limit so that the crl module could opt-in to reading DER values larger than the implicit maximum supported by ring.

Unfortunately this update wasn't quite complete: we need a der::nested_limited version of der::nested to read the nested sequence with the large signed data we pass to signed_data::parse_signed_data or we'll run into the implicit limit before we get to the parsing that supports a larger data size.

Similarly when reading the SEQUENCE that holds all of the revoked certificates we need to use der::expect_tag_and_get_value_limited with a large limit instead of using der::expect_tag_and_get_value with the implicit limit. This sequence is expected to be quite large when there are a large number of revoked certs.

ctz commented on tests/client_auth_revocation/ee_revoked_badsig.crl.der:
I think most/all of these binary changes are because the generation script is (unfortunately) not deterministic, so public keys/signatures differ each time. Often the only changes needing commit are altered tests plus the .rs file, and other changes can be reset to produce less churn.

djc replied:
Yeah, it would sure be nice to avoid churning all of this stuff every time the tests need to be regenerated. Ideally that would also make it feasible to have the test regeneration in the same commit as the changes that require test regeneration.

cpu replied:
That sounds worth pursuing. Maybe the quickest solution is to have `generate.py` not overwrite existing cert/keypairs unless given some kind of `--force` flag. I'll give that a shot. I think that will greatly reduce the churn.

cpu replied:
(and we'd run the CI task for `generate.py` with `--force` enabled I think)

cpu replied:
https://github.com/rustls/webpki/pull/93

cpu commented:
Rebased. Because I removed/renamed some test cases I unfortunately did have to do the noisy binary test file regeneration dance for the crl revocation tests as part of this branch. Hopefully the last time that's needed :crossed_fingers:

djc commented on src/der.rs:
```diff
@@ -77,6 +77,20 @@ pub(crate) fn expect_tag_and_get_value_limited<'a>(
+    size_limit: usize,
+) -> Result<R, E>
+where
+    F: FnOnce(&mut untrusted::Reader<'a>) -> Result<R, E>,
```
Recommend using `impl FnOnce(..) -> ..` for the `decoder` argument directly instead to make this more concise and avoid the type argument indirection.

cpu replied:
Done. Thanks

cpu commented:
> verify_cert: remove CrlProvider, pass CRLs directly.

I think we're going in a different direction. Going to close this and split the other commits into a new non-draft PR that can be merged as-is while I rework this part of the interface/impl.
